aom/src/main/java/com/nedap/archie/aom/primitives/CString.java

```diff
@@ -115,13 +115,9 @@ private boolean hasMatchingConstraint(String constraint, CString otherString) {
         for(String otherConstraint:otherString.constraint) {
             boolean otherIsRegexp = CString.isRegexConstraint(otherConstraint);
             if(otherIsRegexp && !isRegexp) {
-
                 if(matchesRegexp(constraint, otherConstraint)) {
                     return true;
                 }
-            } else if (otherIsRegexp && isRegexp) {
-                //this is very hard to validate. Don't allow it until the spec gets updated to allow this
-                return false;
```

Comment:
Because it's not necessary. If both are regexpes and they are different, then it will eventually go to the `return false` clause at the bottom of the method.
if the child is a regexp, it will now only return true if the parent has the exact same regexp. That case returned false before, and I don't think it should.
This could benefit for a new validity test archetype I guess